app/apollo/resolvers/channel.js

```diff
@@ -478,18 +491,22 @@ const channelResolvers = {
 
       const user = whoIs(me);
 
-      try{
-        logger.info({ req_id, user, org_id, uuid, name }, `${queryName} validating`);
+      try {
+        logger.info({req_id, user, org_id, uuid, name}, `${queryName} validating`);
 
         validateString( 'org_id', org_id );
         validateString( 'uuid', uuid );
         validateName( 'name', name );
 
+        await validAuth(me, org_id, ACTIONS.UPDATE, TYPES.CHANNEL, queryName, context, [uuid, name]);
+        logger.info({req_id, user, org_id, uuid, name}, `${queryName} validating - authorized`);
+
         const channel = await models.Channel.findOne({ uuid, org_id });
-        if(!channel){
+        logger.info({req_id, user, org_id, uuid, name}, `${queryName} validating - found: ${!!channel}`);
+
+        if (!channel) {
           throw new NotFoundError(context.req.t('Channel uuid "{{channel_uuid}}" not found.', {'channel_uuid':uuid}), context);
         }
-        await validAuth(me, org_id, ACTIONS.UPDATE, TYPES.CHANNEL, queryName, context, [channel.uuid, channel.name]);
 
```

Comment:
The passed `name` arg is the _new_ name for the Channel.  To correctly check authorization on this action:
```
        // Get the existing Channel by uuid
        const channel = await models.Channel.findOne({ uuid, org_id });
        logger.info({req_id, user, org_id, uuid, name}, `${queryName} validating - found: ${!!channel}`);

        // Verify user is authorized on the existing Channel
        const identifiers = channel ? [uuid, channel.name] : [uuid];
        await validAuth(me, org_id, ACTIONS.UPDATE, TYPES.CHANNEL, queryName, context, identifiers);
        logger.info({req_id, user, org_id, uuid, name}, `${queryName} validating - authorized`);
        
        //Throw error if user is authorized, but the existing Channel could not be found
        if (!channel) {
          throw new NotFoundError(context.req.t('Channel uuid "{{channel_uuid}}" not found.', {'channel_uuid':uuid}), context);
        }

        // Verify user will STILL be authorized on the Channel if it is renamed -- the user should be prevented from renaming if they only have authorization based on the original name
        const newIdentifiers = [uuid, name];
        await validAuth(me, org_id, ACTIONS.UPDATE, TYPES.CHANNEL, queryName, context, newIdentifiers);

```

Reply:
fixed

Reply:
fixed

Reply:
Unresolving, not addressed

Reply:
I'm not sure why the diff isn't updated here, but pulling down the pr to my local code it shows I have this exact change already made. Not sure why the diff is off...

Reply:
I think the fault was on my end -- refreshing a diff between specific commits rather than getting the overall diff for the PR fresh.
Resolved.